content/en/docs/languages/java/performance.md

```diff
@@ -0,0 +1,186 @@
+increase agent overhead. For more information on how to turn off unnecessary
+instrumentations, see
+[Turn off specific instrumentations](#turn-off-specific-instrumentations).
+
+> [!NOTE] Experimental features of the Java agent might increase agent overhead
+> due to the experimental focus on functionality over performance. Stable
+> features are safer in terms of agent overhead.
+
+## Troubleshooting agent overhead issues
+
+When troubleshooting agent overhead issues, do the following:
+
+- Check minimum requirements. See
+  [Prerequisites](/docs/languages/java/getting-started/#prerequisites).
+- Use the latest compatible version of the Java agent.
+- Use the latest compatible version of your JVM.
+
+Consider taking the following actions to decrease agent overhead:
+
+- If your application is approaching memory limits, consider giving it more
+  memory.
+- If your application is using all the CPU, you might want to scale it
+  horizontally.
+- Try turning off or tuning metrics.
+- Tune trace sampling settings to reduce span volume.
+- Turn off specific instrumentations.
+- Review manual instrumentation for unnecessary span generation.
+
+## Guidelines for measuring agent overhead
+
+Measuring agent overhead in your own environment and deployments provides
+accurate data about the impact of instrumentation on the performance of your
+application or service. The following guidelines describe the general steps for
+collecting and comparing reliable agent overhead measurements.
+
+### Decide what you want to measure
+
+Different users of your application or service might notice different aspects of
+agent overhead. For example, while end users might notice degradation in service
+latency, power users with heavy workloads pay more attention to CPU overhead. On
+the other hand, users who deploy frequently, for example due to elastic
+workloads, care more about startup time.
+
+Reduce your measurements to factors that are sure to impact user experience, so
+your datasets don't contain irrelevant information. Some examples of
+measurements include the following:
+
+- User average, user peak, and machine average CPU usage
+- Total memory allocated and maximum heap used
+- Garbage collection pause time
+- Startup time in milliseconds
+- Average and percentile 95 (p95) service latency
+- Network read and write average throughput
+
+### Prepare a suitable test environment
+
+By measuring agent overhead in a controlled test environment you can better
+identify the factors affecting performance. When preparing a test environment,
+complete the following:
+
+1.  Make sure that the configuration of the test environment resembles
+    production.
+2.  Isolate the application under test from other services that might interfere.
+3.  Turn off or remove all unnecessary system services on the application host.
+4.  Ensure that the application has enough system resources to handle the test
+    workload.
+
+### Create a battery of realistic tests
+
```

Comment:
just a thought but would it make sense to provide a general disclaimer/warning about being careful about running up bills with observability backends when designing and running these tests? For example a large scale load test to determine the overhead but with a sample rate of 100%, might be fine for some companies but could also provide a shock bill if they have a tight budget and aren't prepared.

Reply:
Good point! I'll add that.

Reply:
How about this wording?

```
{{% alert title="Note" color="info" %}}

Tests might increase costs when targeting observability backends and other
commercial services. Plan your tests accordingly or consider using alternative
solutions, such as self-hosted or locally run backends.

{{% /alert %}}
```